aws-ecs/30-bastion.tf

```
@@ -0,0 +1,75 @@
+data "aws_ami" "amazon_linux_2" {
+  most_recent = true
+  filter {
+    name   = "owner-alias"
+    values = ["amazon"]
+  }
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm*"]
+  }
+  owners = ["amazon"]
+}
+
+resource "aws_security_group" "bastion" {
+  count  = var.create_bastion ? 1 : 0
+  name   = "bastion-sg"
+  vpc_id = local.vpc_id
+
+  egress {
+    description = "Allow All 443 Egress for SSM Accessye"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    "Name" = "bastion_sg"
+  }
+}
+
+resource "aws_iam_role" "bastion" {
+  count = var.create_bastion ? 1 : 0
+  name  = "pvault-bastion-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "bastion" {
+  count      = var.create_bastion ? 1 : 0
+  role       = one(aws_iam_role.bastion).name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_instance_profile" "bastion" {
+  count = var.create_bastion ? 1 : 0
+  name  = "vault-bastion-iam-profile"
```

It would be great if you export a variable `var.project_name` or similar, and use it as prefix to all resources:
```suggestion
  name  = "${var.project_name}-bastion-iam-profile"
```

---

Good point. I also suggested using some random id suffix to make it possible to deploy several instances in the same project.